examples/regression/script.jl

```diff
@@ -0,0 +1,196 @@
+# A recreation of <https://gpflow.readthedocs.io/en/master/notebooks/advanced/gps_for_big_data.html>
+
+# # Stochastic Variational Regression
+#
+# In this example, we show how to construct and train the stochastic variational
+# Gaussian process (SVGP) model for efficient inference in large scale datasets.
+# For a basic introduction to the functionality of this library, please refer to
+# the [User Guide](@ref).
+#
+# ## Setup
+
+using SparseGPs
+using Distributions
+using LinearAlgebra
+using IterTools
+
+using Plots
+default(; legend=:outertopright, size=(700, 400))
+
+using Random
+Random.seed!(1234)
+#md nothing #hide
+
+# ## Generate some training data
+#
+# The data generating function `g`
+
+function g(x)
+    return sin(3π * x) + 0.3 * cos(9π * x) + 0.5 * sin(7π * x)
+end
+
+N = 10000 # Number of training points
+x = rand(Uniform(-1, 1), N)
+y = g.(x) + 0.3 * randn(N)
+
+scatter(x, y; xlabel="x", ylabel="y", markershape=:xcross, markeralpha=0.1, legend=false)
+
+# ## Set up a Flux model
+#
+# We shall use the excellent framework provided by [Flux.jl](https://fluxml.ai/)
+# to perform stochastic optimisation. The SVGP approximation has three sets of
+# parameters to optimise - the inducing input locations, the mean and covariance
+# of the variational distribution `q` and the parameters of the
+# kernel.
+#
+# First, we define a helper function to construct the kernel from its parameters
+# (often called kernel hyperparameters), and pick some initial values `k_init`.
+
+using Flux
+
+function make_kernel(k)
+    return softplus(k[1]) * (SqExponentialKernel() ∘ ScaleTransform(softplus(k[2])))
+end
+
+k_init = [0.3, 10]
+#md nothing #hide
+
+# Then, we select some inducing input locations `z_init`. In this case, we simply choose
+# the first `M` data inputs.
+
+M = 50 # number of inducing points
+z_init = x[1:M]
+#md nothing #hide
+
+# Finally, we initialise the parameters of the variational distribution `q(u)`
+# where `u ~ f(z)` are the pseudo-points. We parameterise the covariance matrix
+# of `q` as `C = AᵀA` since this guarantees that `C` is positive definite.
+
+m_init = zeros(M)
+A_init = Matrix{Float64}(I, M, M)
+q_init = MvNormal(m_init, A_init'A_init)
+#md nothing #hide
+
+# Given a set of parameters, we now define a Flux 'layer' which forms the basis
+# of our model.
+
+struct SVGPModel
+    k  # kernel parameters
+    z  # inducing points
+    m  # variational mean
+    A  # variational covariance
+end
+
+Flux.@functor SVGPModel (k, z, m, A)
+#md nothing #hide
+
+# Create the 'model' from the parameters - i.e. return the FiniteGP at inputs x,
+# the FiniteGP at inducing inputs z and the variational posterior over inducing
+# points - q(u).
```

Comment:
This paragraph seems to hang a bit in the middle of nowhere - how does it connect to the previous/next ones?

Reply:
Ah - I forgot to remove this when moving things around